### PR TITLE
Selective oil permeation - zero-order model and flowsheet

### PR DESCRIPTION
**Pull request by tarnold17**
## Fixes/Resolves:

Add zero-order model and flowsheet for selective oil permeation.

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: watertap/examples/flowsheets/case_studies/selective_oil_permeation/selective_oil_permeation.py [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+
+    m.fs.costing.cost_process()
+
+    m.fs.costing.add_electricity_intensity(m.fs.feed.properties[0].flow_vol)
+
+    m.fs.costing.byproduct_oil_volume = Expression(
+        expr=m.fs.costing.utilization_factor
+        * pyunits.convert(
+            m.fs.byproduct_oil.properties[0].flow_vol,
+            to_units=pyunits.bbl / m.fs.costing.base_period,
```
Comment (tarnold17) [inline watertap/examples/flowsheets/case_studies/selective_oil_permeation/selective_oil_permeation.py:173]: Use `pyunits.oil_bbl` instead.

File: watertap/examples/flowsheets/case_studies/selective_oil_permeation/selective_oil_permeation_case_study.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+defined_flows:
+  electricity:
+    value: 0.08
+    units: USD_2020/kWh
+  water:
+    value: 1.0e-12
+    units: USD_2020/m^3
+  oil:
+    value: 80
+    units: USD_2020/bbl
```
Comment (tarnold17) [inline watertap/examples/flowsheets/case_studies/selective_oil_permeation/selective_oil_permeation_case_study.yaml:12]: Use `oil_bbl` instead.

**Review by luohezhiming — APPROVED** [review]
Looks good to me. You could potentially display the flowsheet results using our current AMO framework: `display_metrics_results `and `display_additional_results` instead of `display_costing`. But it is a minor point, if no further ui staff needs to be added, we can keep it.

**Comment by tarnold17** [comment]
The end goal for selective oil permeation is to build a full model. We have determined that not much value would be gained by merging in the zero-order model, so I am closing this PR.

**Pull request closed without merging**